pkg/component/observability/monitoring/persesoperator/rbac.go

```diff
@@ -0,0 +1,109 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package persesoperator
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	clusterRoleName        = "perses-operator"
+	leaderElectionRoleName = "perses-operator-leader-election"
+)
+
+func (p *persesOperator) clusterRole() *rbacv1.ClusterRole {
+	return &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   clusterRoleName,
+			Labels: GetLabels(),
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{appsv1.GroupName},
+				Resources: []string{"deployments", "statefulsets"},
```

Comment:
Why does the `perses-operator` need permission to `Deployment`s AND `StatefulSet`s? Is that a requirement of the operator itself? I would have expected that we would deploy `perses` always as `Deployment` analogous to we do it with `plutono`. We could also use `StatefulSet`s instead, but hopefully we do not use both.
Please clarify.

Reply:
It's in the upstream RBAC configuration, because Perses can indeed be deployed both as a `StatefulSet` and a `Deployment`. I'm almost certain running it as a `Deployment` will be fine for our use case, but I'd rather remove the permission when I know for sure the `perses-operator` won't manage any `Statefulsets`.